File: LeetCode/Easy/728_self_dividing_numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []
        for i in range(left, right + 1):
            num = str(i)
            if '0' in num:
                pass
            else:
                for j in range(len(num)):
                    if int(num) % int(num[j]) != 0:
                        break
                    elif j == len(num) - 1 and int(num) % int(num[j]) == 0:
                        res.append(int(num))
        return res

File: LeetCode/Java/Easy/728_self_dividing_numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> ans = new ArrayList<>();
        for (int i = left; i <= right; i++)
            if (isValid(i))
                ans.add(i);
        return ans;
    }
    
    public boolean isValid(int num) {
        int curr = num;
        while(curr != 0){
            int digit = curr % 10;
            if (digit == 0 || num % digit != 0)
                return false;
            curr /= 10;
        }
        return true;
    }
}